tests/integration_test.py

```diff
@@ -102,3 +102,22 @@ def test_runenv_shell(shell, not_found_message, built_rustenv, tmpdir):
 
     assert stages['installed']['hello'] == 'Hello World!'
     assert stages['uninstalled'] == stages['activated']
+
+
+@pytest.mark.parametrize(
+    ('shell'),
+    (
+        (
+            'bash',
+        ),
+        (
+            'zsh',
+        ),
+    ),
+)
+def test_works_without_ps1(shell, built_rustenv, tmpdir):
+    test_script = tmpdir.join('test.sh')
+    test_script.write('unset PS1\n')
+    test_script.write('set -euo pipefail\n', mode='a')
+    test_script.write(f'. {built_rustenv.strpath}/bin/activate\n', mode='a')
```

Comment:
I would just write one big string rather than using the (somewhat obscure) `mode='a'`